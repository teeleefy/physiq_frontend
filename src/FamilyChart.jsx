import { Button, Card, CardBody, CardTitle, CardText, ListGroup } from "reactstrap";
import { NavLink } from "react-router-dom";
import { useState, useContext, useEffect } from "react";
import {FamilyContext} from "./auth/UserContext";
import PhysiqApi from "./Api.js";
import './styles/FamilyChart.css'

function FamilyChart(){
    let { currentFamily } = useContext(FamilyContext);

    function renderFamilyChart(){
        if(currentFamily){
            return (
                <>
                 <CardTitle className="text-center">
                    <h1 className="Home-title">Welcome to fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b>!</h1>
                </CardTitle>
                <CardText>
                    <h2>First off, what is fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b>? üëÄ</h2>
                    <p>That's simple! It is an easy-to-use app that allows you to chart on yourself and your loved ones!</p>
                    <h2>...And why would you want to do that? üßê</h2>
                    <p>When it comes to managing your family's health care, there are a <b>LOT</b> of little details to keep track of! <b>When</b> was your child's last dentist appointment? <b>Who</b> was that doctor you saw last fall before you moved states? <b>What</b> is the name of that medication that doctor prescribed you, and <b>why</b> do you even take it?  <b>It's a lot to keep up with!</b></p>
                    <h2>That's where fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b> comes in to play! üí™</h2>
                    <p>Here, you can add yourself and your family members to your group, connect to each member's personal portal, and add, edit, and delete their personal health-related details.  This will be a very helpful tool for you and your family as those details pile up over the years! After all, Nurses and Doctors keep records of your health journey... <b>Why can't you?</b></p>
                    <h2>Sounds great! How do I get started?? üëè</h2> 
                    <p><b>First,</b> add a member to your group in your Home Dashboard. &nbsp;<b>Next,</b> click their name to transport to their Member Portal. <b>From there,</b> navigate through the different tabs to add allergies, diagnoses, etc.  &nbsp;<b>Later,</b> you can easily update or delete these details from their respective pages in the Member Portal!</p>
                    <NavLink to="/home" >
                <Button className="FamilyChart-btn">Let's Get Started!</Button>
            </NavLink>
                </CardText>
                </>
            )
        }
        return (
            <div className="Home-btns">
            <CardTitle className="text-center">
                <h1 className="Home-title">Welcome to fa<b className='mychart'>m</b>il<b className='mychart'>y</b> <b className='mychart'>chart</b>!</h1>
            </CardTitle>
            <CardText className="Home-text">
                Keep track of your family's health, all in one, convenient place!
            </CardText>
            <NavLink to="/login" >
                <Button className="FamilyChart-btn">Login</Button>
            </NavLink>
           <NavLink to="/signup" >
                <Button className="FamilyChart-btn justify-content-center">Sign Up</Button>
            </NavLink>
            </div>
        )
    }


 return(
    <div id="FamilyChart-main">
        <div id="FamilyChart-message">
            {renderFamilyChart()} 
        </div>  
    </div>
 )
}

export default FamilyChart;