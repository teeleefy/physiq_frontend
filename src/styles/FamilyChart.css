#FamilyChart-main {
    width: 90vw;
    height: 82vh;
    background-color: white;
    overflow: scroll;
    color: black;
    border: 15px red solid;
    border-radius: 15px;
    padding: 0;
    margin: 0;
    text-align: center;
  }

#FamilyChart-message{
    width: 55vw;
    margin: auto;
}

#FamilyChart-message p {
    font-size: 1.2rem;
}

.FamilyChart-title{
    margin-bottom: 1rem;
    font-weight: 600;
    text-shadow: 2px 2px 2px grey;
    font-size: 3rem; 
}

.FamilyChart-text {
    font-size: 1.2rem; 
}

.FamilyChart-btns{
    text-align: center;
}

.FamilyChart-btn{
    color: black !important;
    text-decoration: none;
    font-size: 2.5rem !important;
    font-weight: bold !important;
    transition: color 0.5s;
    margin: 1rem; 
    padding: 1rem;
    background-color: red !important;
    box-shadow: 5px 5px 8px black;
    border:0 !important;
    width: 20vw !important;
    transition: all 0.6s !important;
}

.FamilyChart-btn:hover{
    background-color: black !important;
    box-shadow: 5px 5px 8px red;
    width: 23vw !important;
    font-size: 2.7rem !important;
    color:red !important;
}

.FamilyChart-btn-welcome{
    color: black !important;
    text-decoration: none;
    font-size: 1.5rem !important;
    font-weight: bold !important;
    transition: color 0.5s;
    margin: auto; 
    background-color: red !important;
    box-shadow: 5px 5px 8px black;
    border:0 !important;
    width: 15vw !important;
    transition: all 0.6s !important;
    text-align: center;
}

.FamilyChart-btn-welcome:hover{
    background-color: black !important;
    box-shadow: 5px 5px 8px red;
    width: 18vw !important;
    font-size: 2rem !important;
    color:red !important;
}

.FamilyChart-addMember{
    color: white !important;
    font-size: 1.5rem;
    text-decoration: none;
    width:12vw;
}

.FamilyChart-familymember{
    background-color: red !important;
    width: 50vw;
    margin: auto;
    border:0 !important;
}