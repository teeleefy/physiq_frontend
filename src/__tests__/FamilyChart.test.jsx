import { renderWithRouterAndFamilyContext, renderWithRouterAndLoggedOutContext, screen } from '../../__tests__/test-tools';
import FamilyChart from '../FamilyChart'

describe('FamilyChart when logged out', () => {
  it('renders the FamilyChart component when logged out', () => {
    renderWithRouterAndLoggedOutContext(<FamilyChart />)
    
    screen.debug(); // prints out the jsx in the FamilyChart component unto the command line
  })

  it("matches snapshot when logged out", function() {
    const { asFragment } = renderWithRouterAndLoggedOutContext(<FamilyChart />);
    expect(asFragment()).toMatchSnapshot();
  });
})

describe('FamilyChart with Family Context', () => {
  it('renders the FamilyChart component', () => {
    renderWithRouterAndFamilyContext(<FamilyChart />)
    
    screen.debug(); // prints out the jsx in the FamilyChart component unto the command line
  })

  it("matches snapshot", function() {
    const { asFragment } = renderWithRouterAndFamilyContext(<FamilyChart />);
    expect(asFragment()).toMatchSnapshot();
  });
})


