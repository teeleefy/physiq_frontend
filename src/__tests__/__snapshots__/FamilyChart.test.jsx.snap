// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FamilyChart when logged out > matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    id="FamilyChart-main"
  >
    <div
      id="FamilyChart-message"
    >
      <div
        class="Home-btns"
      >
        <div
          class="text-center card-title"
        >
          <h1
            class="Home-title"
          >
            Welcome to fa
            <b
              class="mychart"
            >
              m
            </b>
            il
            <b
              class="mychart"
            >
              y
            </b>
             
            <b
              class="mychart"
            >
              chart
            </b>
            !
          </h1>
        </div>
        <p
          class="Home-text card-text"
        >
          Keep track of your family's health, all in one, convenient place!
        </p>
        <a
          class=""
          href="/login"
        >
          <button
            class="FamilyChart-btn btn btn-secondary"
          >
            Login
          </button>
        </a>
        <a
          class=""
          href="/signup"
        >
          <button
            class="FamilyChart-btn justify-content-center btn btn-secondary"
          >
            Sign Up
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FamilyChart with Family Context > matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="FamilyChart-main"
  >
    <div
      id="FamilyChart-message"
    >
      <div
        class="text-center card-title"
      >
        <h1
          class="Home-title"
        >
          Welcome to fa
          <b
            class="mychart"
          >
            m
          </b>
          il
          <b
            class="mychart"
          >
            y
          </b>
           
          <b
            class="mychart"
          >
            chart
          </b>
          !
        </h1>
      </div>
      <p
        class="card-text"
      />
      <h2>
        First off, what is fa
        <b
          class="mychart"
        >
          m
        </b>
        il
        <b
          class="mychart"
        >
          y
        </b>
         
        <b
          class="mychart"
        >
          chart
        </b>
        ? 👀
      </h2>
      <p>
        That's simple! It is an easy-to-use app that allows you to chart on yourself and your loved ones!
      </p>
      <h2>
        ...And why would you want to do that? 🧐
      </h2>
      <p>
        When it comes to managing your family's health care, there are a 
        <b>
          LOT
        </b>
         of little details to keep track of! 
        <b>
          When
        </b>
         was your child's last dentist appointment? 
        <b>
          Who
        </b>
         was that doctor you saw last fall before you moved states? 
        <b>
          What
        </b>
         is the name of that medication that doctor prescribed you, and 
        <b>
          why
        </b>
         do you even take it?  
        <b>
          It's a lot to keep up with!
        </b>
      </p>
      <h2>
        That's where fa
        <b
          class="mychart"
        >
          m
        </b>
        il
        <b
          class="mychart"
        >
          y
        </b>
         
        <b
          class="mychart"
        >
          chart
        </b>
         comes in to play! 💪
      </h2>
      <p>
        Here, you can add yourself and your family members to your group, connect to each member's personal portal, and add, edit, and delete their personal health-related details.  This will be a very helpful tool for you and your family as those details pile up over the years! After all, Nurses and Doctors keep records of your health journey... 
        <b>
          Why can't you?
        </b>
      </p>
      <h2>
        Sounds great! How do I get started?? 👏
      </h2>
      <p>
        <b>
          First,
        </b>
         add a member to your group in your Home Dashboard.  
        <b>
          Next,
        </b>
         click their name to transport to their Member Portal. 
        <b>
          From there,
        </b>
         navigate through the different tabs to add allergies, diagnoses, etc.   
        <b>
          Later,
        </b>
         you can easily update or delete these details from their respective pages in the Member Portal!
      </p>
      <a
        class=""
        href="/home"
      >
        <button
          class="FamilyChart-btn btn btn-secondary"
        >
          Let's Get Started!
        </button>
      </a>
      <p />
    </div>
  </div>
</DocumentFragment>
`;
